dist/functions.setup-appliance.sh

```diff
@@ -239,9 +239,17 @@ function prepare_database_setup {
 
   if [[ $? > 0 ]];then
     echo "Initialize MySQL databases (first time only)"
-    DATADIR_FILE=$(grep datadir -rl /etc/my.cnf*)
-    echo " - reconfiguring datadir in $DATADIR_FILE"
-    perl -p -i -e 's#.*datadir\s*=\s*/var/lib/mysql$#datadir= /srv/obs/MySQL#' $DATADIR_FILE
+
+    echo " - writing new datadir to /etc/my.cnf.d/obs-server.cnf"
+    cat <<EOF > /etc/my.cnf.d/obs-server.cnf
+[mysqld]
+bind-address = 127.0.0.1
+datadir = /srv/obs/MySQL
+
+[mysqld_multi]
+datadir = /srv/obs/MySQL
+EOF
+
```

Comment:
Could we combine both approaches? First do a `grep` and if we found nothing, create obs-server.cnf - otherwise reconfigure the existing entries?

Reply:
I've dropped the patch that creates `obs-server.cnf` and used sed to replace only the first occurrence of `datadir` in `/etc/my.cnf`